### PR TITLE
SAK-43678 Assignments: Incorrect order in name column when downloading spreadsheet [XLSX]

### PR DESCRIPTION
**Pull request by joaquinmarques**
@ern I can't use directly org.sakaiproject.util.comparator.UserSortNameComparator because this comparator is for User class and my class is Submitter, so I can't do this:

submitters.sort(new UserSortNameComparator());

The error:

The method sort(Comparator<? super GradeSheetExporter.Submitter>) in the type List<GradeSheetExporter.Submitter> is not applicable for the arguments (UserSortNameComparator)

What would be the correct way to do this?

### CODE REVIEW

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/GradeSheetExporter.java [hunk 1]
```diff
@@ -70,7 +74,28 @@
...
+                this.collator = new RuleBasedCollator(
+                        ((RuleBasedCollator) this.collator).getRules().replaceAll("<'\u005f'", "<' '<'\u005f'"));
+            } catch (final ParseException e) {
+                log.warn(this + " Cannot init RuleBasedCollator. Will use the default Collator instead.", e);
+            }
+        }
 
+        @Override
+        public int compare(final Submitter s1, final Submitter s2) {
+            return this.collator.compare(s1.sortName, s2.sortName);
```
Comment (ern) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/GradeSheetExporter.java:95]: Can produce an NPE either use a lambda to filter nulls or the NullSafeComparator i.e.
```return new NullSafeComparator<>(collator, false).compare(s1.getSortName(), s2.getSortName()```

**Review by ern — CHANGES_REQUESTED** [review]
@joaquinmarques just need to handle nulls see my comment and suggestion

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/GradeSheetExporter.java (hunk 1 above)
Comment (joaquinmarques) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/GradeSheetExporter.java]: Done!

**Comment by danielmerino** [comment]
@ern Does the new code satisfy your requested changes? Thanks for your review.

**Review by ern — APPROVED** [review]

**Pull request merged**